File: JAVA/Max sum in the configuration.java
class Solution {
    long max_sum(int a[], int n) {
        long s=0,cs=0;
        long ind;
        for(int i=0;i<n;i++){
            s+=a[i];
            ind=i;
            cs+=ind*a[i];
        }
        long ans=cs;
       int k=n-1;
       long N=n;
       while(k>0){
           cs=cs+s-(N*a[k]);
           ans=Math.max(ans,cs);
           k--;
       }
        return ans;
    }
}

File: C++/Max sum in the configuration.cpp
class Solution {
  public:
    long long max_sum(int a[], int n) {
         long long sum = 0, product_sum = 0, ans = 0, N = n;
        for(int i = 0; i<n; i++){
            sum += a[i];
            product_sum += i*(long long)a[i];
        }
        ans = product_sum;
        for(int i = 1; i<n; i++){
            product_sum = (product_sum - sum) + a[(i + n - 1)%n]*(N);
            ans = max(ans, product_sum);
        }
        return ans;
    }
};
